app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -230,23 +238,43 @@ def global_content_counts
       end
 
       def sync_container_gateway
-        if has_feature?(::SmartProxy::CONTAINER_GATEWAY_FEATURE)
+        return unless has_feature?(::SmartProxy::CONTAINER_GATEWAY_FEATURE)
+
+        begin
           update_container_repo_list
+        rescue StandardError => e
+          Rails.logger.error("Failed to update container repository list for #{name}: #{e.message}")
+          Rails.logger.debug(e.backtrace.join("\n"))
+        end
+
+        begin
           users = container_gateway_users
           update_user_container_repo_mapping(users) if users.any?
-          if content_facets.any?
-            update_container_gateway_hosts
-            update_host_container_repo_mapping(content_facets)
+        rescue StandardError => e
+          Rails.logger.error("Failed to update user container repository mapping for #{name}: #{e.message}")
+          Rails.logger.debug(e.backtrace.join("\n"))
+        end
+
+        begin
+          facets = subscription_facets.to_a
```

Comment:
We wouldn't want to do `.to_a` on thousands of facets and load them all into memory just so we can use `.any?`. I think Active Record has a better way (`.exists?` I think?)